### PR TITLE
WIP: Defer last commit info

### PR DESCRIPTION
**Pull request by zeripath**
We use the commitinfo cache because we recognise that generating its data is slow. However, when we have a commit info cache we allow it to cause the page to not be rendered until the commit info has been calculated.

This PR simply changes the code to allow the page to still be rendered at least partially whilst we're waiting.

### CODE REVIEW

File: modules/repository/last_commit_queue_gogit.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
+// Copyright 2019 Gitea. All rights reserved.
```
Comment (techknowlogick) [inline modules/repository/last_commit_queue_gogit.go]:
```suggestion
// Copyright 2021 The Gitea Authors. All rights reserved.
```

File: modules/repository/last_commit_queue_nogogit.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
+// Copyright 2019 Gitea. All rights reserved.
```
Comment (techknowlogick) [inline modules/repository/last_commit_queue_nogogit.go]:
```suggestion
// Copyright 2021 The Gitea Authors. All rights reserved.
```

**Review by techknowlogick — COMMENTED** [review]
tiniest of nits. Still going through the other code though.

**Comment by zeripath** [comment]
Things to be aware of:

The use of a queue with the default workers of 0 and boost of 1 with queue length 20 means that there is only 1 repository getting its commit info generated at a time unless there are 20 waiting. We may want to increase the number of boost workers for this queue by default.

**Comment by zeripath** [comment]
In light of information found in #16035 I think I'm going to have to make this WIP and have another think.

A different solution is required.

**Comment by tuaris** [comment]
@zeripath I am wondering what prompted you to rethink this?  Unless I am misunderstanding the proposed solution, #16035 is irrelevant?

The way I understand the approach is:

1. Remove the 'on the fly' extraction of the commit message via git log during page rendering.
2. Instead look for the commit message as a value in a SQL database, Elasticsearch, or this new Ledis thing (I don't believe using Redis or Memcache make sense here).
3. If there is no value display nothing (perfectly acceptable behavior).
4. A background process will fill/update/delete the SQL/Elasticsearch values.

File: modules/setting/queue.go [hunk 3]
```diff
@@ -146,6 +146,16 @@ func NewQueueService() {
 		_, _ = section.NewKey("CONN_STR", Indexer.IssueQueueConnStr)
 	}
 
+	// Default the last_commit_queue to use the level queue
+	section = Cfg.Section("queue.issue_indexer")
```
Comment (lunny) [inline modules/setting/queue.go:150]: the queue name is not right.

**Comment by delvh** [comment]
Can this PR be closed then as #16467 got merged now?

**Comment by zeripath** [comment]
@delvh whilst this does do things differently and does some novel things I think you're right and I'm gonna close it.

**Pull request closed without merging**